src/redisraft.c

```diff
@@ -757,7 +764,14 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
         return;
     }
 
-    RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
+    long long timeout = 0;
+    if (cmd_flags & CMD_SPEC_BLOCKING) { /* protect against blocking commands in a MULTI above */
+        if (RaftRedisExtractBlockingTimeout(ctx, cmds, &timeout) != RR_OK) {
+            return;
+        }
+    }
+
+    RaftReq *req = RaftReqInitBlocking(ctx, RR_REDISCOMMAND, timeout);
```

Comment:
Why no to run `RaftReqInitBlocking()` only for `CMD_SPEC_BLOCKING` case?

Reply:
that's a mistake on my part.

Reply:
code changed to 

```
    RaftReq *req;
    if (cmd_flags & CMD_SPEC_BLOCKING) { /* protect against blocking commands in a MULTI above */
        long long timeout = 0;
        if (RaftRedisExtractBlockingTimeout(ctx, cmds, &timeout) != RR_OK) {
            return;
        }
        req = RaftReqInitBlocking(ctx, RR_REDISCOMMAND, timeout);
    } else {
        req = RaftReqInit(ctx, RR_REDISCOMMAND);
    }
    RaftRedisCommandArrayMove(&req->r.redis.cmds, cmds);
```